watertap/unit_models/zero_order/uv_aop_zo.py

```diff
@@ -0,0 +1,130 @@
+
+        if self.config.process_subtype is None:
+            raise ConfigurationError(
+                f"{self.name} - zero-order uv/aop operations "
+                "require the process_subtype configuration argument to be set")
+
+        build_siso(self)
+        constant_intensity(self)
+
+        self.uv_reduced_equivalent_dose = Var(self.flowsheet().time,
+                                              units=pyunits.mJ/pyunits.cm**2,
+                                              doc="Reduced equivalent dosage")
+        self.uvt_in = Var(self.flowsheet().time,
+                          units=pyunits.dimensionless,
+                          doc="UV transmittance of solution at UV reactor inlet")
+
+        self._fixed_perf_vars.append(self.uv_reduced_equivalent_dose)
+        self._fixed_perf_vars.append(self.uvt_in)
+
+
+        if self.config.has_aop:
+            self.oxidant_dose = Var(self.flowsheet().time,
+                                    units=pyunits.mg/pyunits.L,
+                                    doc="Oxidant dosage")
+
+            self.chemical_flow_mass = Var(
+                self.flowsheet().time,
+                units=pyunits.kg/pyunits.s,
+                bounds=(0, None),
+                doc="Mass flow rate of oxidant solution")
+
+
+            self._fixed_perf_vars.append(self.oxidant_dose)
+
+            @self.Constraint(self.flowsheet().time,
+                             doc="Chemical mass flow constraint")
+            def chemical_flow_mass_constraint(b, t):
+                return (b.chemical_flow_mass[t] ==
+                        pyunits.convert(b.oxidant_dose[t]
+                        * b.properties_in[t].flow_vol, to_units=pyunits.kg/pyunits.s))
+
+            self._perf_var_dict["Oxidant Dosage (mg/L)"] = self.oxidant_dose
+            self._perf_var_dict["Oxidant Flow (kg/s)"] = self.chemical_flow_mass
+
+    def load_parameters_from_database(self, use_default_removal=False):
+        """
+                Method to load parameters from database, with modification for UV-AOP.
+
+                Args:
+                    use_default_removal - (optional) indicate whether to use defined
+                                          default removal fraction if no specific value
+                                          defined in database
+
+                Returns:
+                    None
+                """
+        # Get parameter dict from database
+        if self._tech_type is None:
+            raise NotImplementedError(
+                f"{self.name} derived zero order unit model has not "
+                f"implemented the _tech_type attribute. This is required "
+                f"to identify the database file to load parameters from.")
+
+        # Get parameter dict from database
+        pdict = self.config.database.get_unit_operation_parameters(
+            self._tech_type, subtype=self.config.process_subtype)
+
+        if not self.config.has_aop:
```

Comment:
I am not sure you need to overload this method. Why not just conditionally add this oxidation dose vars to `._perf_var_dict` for the AOP case?

Reply:
I conditionally added oxidation dose in the build. The reason for this overload was only to raise an exception when the user keeps `has_aop=False` but chooses a subtype that _does_ include aop. In that case, the chemical flow and dose vars would not be created, and chemical flow wouldn't be calculated.

Reply:
Wouldn't the base method have raised an exception anyway?

Reply:
No, it doesn't. It does raise it when `has_aop` is True and the user picks a subtype without aop. This is because when attempting to fix `oxidant_dose`, no such parameter would be found in the subtype.

When `has_aop` is False and the user picks a subtype with aop, the build doesn't create `oxidant_dose` and therefore never attempts to fix an oxidant_dose parameter, and `load_parameters_from_database` doesn't expect to find `oxidant_dose`. Thus, no exception is raised.

Reply:
OK, that I think is the concrete argument for having two separate models and two separate YAML files.

The data for systems with AOP is only valid in the case where AOP is present (i.e. the removal fractions are assuming both the effects of UV and oxidant), thus I don't think it should be combined with those cases where it is not. The AOP model can still inherit from the UV model, but I think they should be separate classes with separate YAML files (and thus technology type). Otherwise, the line between the two gets too blurred.

Note that in other applications with subtypes, the model form is the same across all subtypes, and only the parameter values change, which is not the case here.